test/suite/test_prepare_lookaside02.py

```diff
@@ -0,0 +1,174 @@
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# test_prepare_lookaside02.py
+#   Prepare updates can be resolved for both commit / rollback operations.
+#
+
+from helper import copy_wiredtiger_home
+import random
+from suite_subprocess import suite_subprocess
+import wiredtiger, wttest
+from wtscenario import make_scenarios
+
+def timestamp_str(t):
+    return '%x' % t
+
+class test_prepare_lookaside02(wttest.WiredTigerTestCase, suite_subprocess):
+    tablename = 'test_prepare_cursor'
+    uri = 'table:' + tablename
+    txn_config = 'isolation=snapshot'
+
+    types = [
+        ('col', dict(extra_config=',log=(enabled=false),key_format=r')),
+        ('row', dict(extra_config=',log=(enabled=false)')),
+        #('lsm', dict(extra_config=',log=(enabled=false),type=lsm')),
+    ]
+
+    # Transaction end types
+    txn_end = [
+        ('txn_commit', dict(txn_commit=True)),
+        ('txn_rollback', dict(txn_commit=False)),
+    ]
+
+    scenarios = make_scenarios(types, txn_end)
+    conn_config = 'log=(enabled)'
+
+    def test_prepare_conflict(self):
+        if not wiredtiger.timestamp_build():
+            self.skipTest('requires a timestamp build')
+
+        self.session.create(self.uri,
+            'key_format=i,value_format=i' + self.extra_config)
+        c = self.session.open_cursor(self.uri)
+
+        # Insert keys 1..100 each with timestamp=key, in some order
+        orig_keys = range(1, 101)
+        keys = orig_keys[:]
+        random.shuffle(keys)
+
+        # Scenario: 1
+        # Check insert operation
+        self.session.begin_transaction(self.txn_config)
+        c[1] = 1
+        # update the value with in this txn
+        self.session.prepare_transaction('prepare_timestamp=' + timestamp_str(100))
+        if self.txn_commit == True:
+            self.session.commit_transaction('commit_timestamp=' + timestamp_str(101))
+        else:
+            self.session.rollback_transaction()
+
+        self.conn.set_timestamp('stable_timestamp=' + timestamp_str(150))
```

Comment:
Do we need setting stable/oldest timestamp and checkpointing in this test. If it isn't functionally required, my preference would be to remove it to keep the test simple.